docs/concepts/realms.md

```diff
@@ -3,47 +3,196 @@ id: realms
+
+As opposed to [packages](./packages.md), realms are stateful, meaning they keep
+their state between transactions calls. In practice, global variables in the
+realm code are automatically persisted after a transaction has been executed,
+resulting in the fact that developers do not need to bother with the intricacies 
+of state management and persistence, like they do with other languages.
+
+### On-chain paths
+Since Gno.land is built for full transparency and auditability, all on-chain Gno
+code is open-sourced. You can view realm code by simply going to its path in
+your web browser. For example, to take a look at the `gno.land/r/demo/users` realm,
+used for user registration, by visiting
+[`gno.land/r/demo/users`](https://gno.land/r/demo/users/users.gno).
+
+:::info
+Depending on the network, the realm domain might change. Currently,
+the `gno.land/` domain (and all of its subdomains, such as `r/`) is pointing to
+the [Portal Loop](./portal-loop.md) testnet endpoint, which is subject
+to change. To view realms on the `test3` network (depr.), prepend `test3` to
+the domain: [`test3.gno.land/r/demo/users`](https://test3.gno.land/r/demo/users).
+:::
+
+[//]: # (Learn more about package paths & allowed namespaces [here].)
+
+To learn how to write a realm, see [How to write a simple Gno Smart Contract](../how-to-guides/simple-contract.md).
+
+## Externally Owned Accounts (EOAs)
+EOAs, or simply `user realms`, are Gno addresses generated from a BIP39 mnemonic
+phrase in a key management application, such as
+[`gnokey`](../gno-tooling/cli/gnokey.md), and [Adena](https://adena.app).
+
+Currently, EOAs are the only realms that can initiate a transaction. They can do
+this by calling any of the possible messages in Gno.land, such as 
+[Call](../gno-tooling/cli/gnokey.md#call),
+[AddPackage](../gno-tooling/cli/gnokey.md#addpkg),
+[Send](../gno-tooling/cli/gnokey.md#send), or Run.
+
+## Working with realms
+In Gno, each transaction consists of a call stack, which is made up of `frames`.
+A single frame is a unique realm in the call stack. Every frame and its properties 
+can be accessed via different functions defined in the `std` package in Gno:
+- `std.GetOrigCaller()` - returns the address of the original signer of the
+transaction
+- `std.PrevRealm()` - returns the previous realm instance, which can be a user realm
+or a smart contract realm
+- `std.CurrentRealm()` - returns the instance of the realm that has called it
+- `std.CurrentRealmPath()` - shorthand for `std.CurrentRealm().PkgPath()`
+- `std.GetCallerAt()` - returns the n-th caller's address, going back in
+the transaction call stack
+
+Let's look at return values of these functions in two distinct situations:
+1. EOA calling a realm
+2. EOA calling a sequence of realms
+
+### 1. EOA calling a realm
+Take these two actors in the call stack:
+```
+EOA:
+    addr: g1jg8mtutu9khhfwc4nxmuhcpftf0pajdhfvsqf5
+    pkgPath: "" // empty as this is a user realm
```

Comment:
I find it weird you're mentioning package paths at all in the context of user accounts